stream/src/placeholder/instance.rs

```diff
@@ -216,25 +195,10 @@ pub struct Receiver<St: Stream> {
     stream: St,
 
     max_message_size: usize,
-    dialer: bool,
-    iter: u16,
-    seq: u64,
+    nonce: nonce::Info,
 }
 
 impl<St: Stream> Receiver<St> {
-    fn peer_nonce(&mut self) -> Result<Nonce, Error> {
```

Comment:
Its odd that we used to encode bytes between `incrementing seq/iter`. The new approach is much cleaner.